File: pos_bahrain/pos_bahrain/report/stock_report/stock_report.js
// Copyright (c) 2024, 	9t9it and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Stock Report"] = {
	"filters": [
		{
			"fieldname": 'company',
			"label": "The Company Name",
			"fieldtype":"Link",
			"options":"Company",
			"default": frappe.defaults.get_user_default('company')
		},
		{
			"fieldname":"item_code",
			"label":"Item Code",
			"fieldtype":"Link",
			"options":"Item",

		},
		{
			"fieldname":"warehouse",
			"label":"Warehouse",
			"fieldtype":"Link",
			"options":"Warehouse",
			
		}
	]
};

File: pos_bahrain/pos_bahrain/report/stock_report/stock_report.py
# Copyright (c) 2024, 	9t9it and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
	data = []
	columns = [
		{
			"fieldname":"item",
			"label":"Item Column",
			"fieldtype":"Link",
			"options":"Item"
		},
		{
			"fieldname":"warehouse",
			"label":"Warehouse Column",
			"fieldtype":"Link",
			"options":"Warehouse"
		}
	]
	custom_filters = {}
	if filters.item_code != None:
		custom_filters["item_code"] = filters.item_code
	if filters.warehouse != None:
		custom_filters["warehouse"] = filters.warehouse
	stock_ledger = frappe.db.get_list("Stock Ledger Entry", filters = custom_filters,  fields= ["*"])
	for stock in stock_ledger:
		data.append({
			"item":stock.item_code,
			"warehouse": stock.warehouse
		})
	# frappe.throw(f"{stock_ledger}")
	return columns, data
